src/core/cs3iface.py

```diff
@@ -169,41 +179,49 @@ def setxattr(endpoint, filepath, userid, key, value, lockmd):
 
 def getxattr(endpoint, filepath, userid, key):
     '''Get the extended attribute <key> using the given userid as access token'''
-    tstart = time.time()
-    reference = _getcs3reference(endpoint, filepath)
-    statInfo = ctx['cs3gw'].Stat(request=cs3sp.StatRequest(ref=reference), metadata=[('x-access-token', userid)])
-    tend = time.time()
-    if statInfo.status.code == cs3code.CODE_NOT_FOUND:
-        log.debug(f'msg="Invoked stat for getxattr on missing file" filepath="{filepath}"')
-        return None
-    if statInfo.status.code != cs3code.CODE_OK:
-        log.error('msg="Failed to stat" filepath="%s" userid="%s" trace="%s" key="%s" reason="%s"' %
-                  (filepath, userid[-20:], statInfo.status.trace, key, statInfo.status.message.replace('"', "'")))
-        raise IOError(statInfo.status.message)
+    ref = _getcs3reference(endpoint, filepath)
+    statInfo = None
+    if _hashedref(endpoint, filepath) not in ctx['xattrcache']:
+        # cache miss, go for Stat and refresh cache
+        tstart = time.time()
```

Comment:
Maybe it would be better to use [`time.perf_counter_ns()`](https://docs.python.org/3/library/time.html#time.perf_counter_ns) here to ensure the highest precision independent on the system running WOPI.